tools/functional-tester/etcd-runner/main.go

```diff
@@ -310,10 +320,24 @@ func runWatcher(eps []string) {
 						return
 					}
 				}
-			}(prefixes[i])
+			}(roundPrefix + "-" + prefixes[i])
 		}
 	}
 	wg.Wait()
+
+	// cancel all watch channels
+	cancelc()
+
```

Comment:
@heyitsanthony i just have one doubt, the behavior should still have been uniform, since all clients were getting closed for watchers, but how come after calling cancel, few channels were coming as closed. Any reason for this?